templates/components/vectordbs/python/none/generate.py

```diff
@@ -15,16 +20,62 @@
 logger = logging.getLogger()
 
 
+def get_vector_store():
+    if not os.path.exists(STORAGE_DIR):
+        vector_store = SimpleVectorStore()
+        return vector_store
+    else:
+        return SimpleVectorStore.from_persist_dir(STORAGE_DIR, namespace="default")
+
+
+def get_doc_store():
+    if not os.path.exists(STORAGE_DIR):
+        docstore = SimpleDocumentStore()
+        return docstore
+    else:
+        return SimpleDocumentStore.from_persist_dir(STORAGE_DIR)
+
+
 def generate_datasource():
     init_settings()
     logger.info("Creating new index")
+
     # load the documents and create the index
     documents = get_documents()
-    index = VectorStoreIndex.from_documents(
-        documents,
+    docstore = get_doc_store()
+    vector_store = get_vector_store()
+    storage_context = StorageContext.from_defaults(
+        vector_store=vector_store,
+        docstore=docstore,
+    )
+
+    # Create ingestion pipeline
+    ingestion_pipeline = IngestionPipeline(
+        transformations=[
+            SentenceSplitter(
```

Comment:
Isn't that using settings values as default?

Reply:
No, i checked. It'll use constant values if we don't explicit specify when initialization, the Setting does not work this case. see: https://github.com/run-llama/llama_index/blob/00f0fea5c0894efb728cab17500dd5d860e31aff/llama-index-core/llama_index/core/node_parser/text/sentence.py#L68C35-L68C35